src/radical/pilot/agent/scheduler/base.py

```diff
@@ -344,19 +352,38 @@ def _change_slot_states(self, slots, new_state):
             #       that we would read, and keep a dictionary that maps the uid
             #       of the node to the location on the list?
 
-            node = (n for n in self.nodes if n['uid'] == node_uid).next()
-            assert(node)
+            node = (n for n in self.nodes if n['uid'] == nodes['uid']).next()
+            # for n in self.nodes:
+            #     if n['uid'] == nodes['uid']:
+            #         node = n
+            #         break
+            # assert(node)
 
```

Comment:
If the list comprehension is too complex, then replace with the verbose version.  If it isn't, then the comment is not needed.

Reply:
This assert is commented out for some reason. I brought it back and removed the comments.